docs/adrs/0000-composite-actions.md

```diff
@@ -0,0 +1,97 @@
+**Date**: 2021-06-10
+
+**Status**: Accepted
+
+## Context
+
+We released [composite run steps](https://github.com/actions/runner/pull/554) last year which started our journey of reusing steps across different workflow files. To continue that journey, we want to expand composite run steps into composite actions.
+
+We want to support the `uses` steps from workflows in composite actions, including:
+  - Container actions
+  - javascript actions
+  - and even other Composite actions (up to a limit of course!)
+  - The pre and post steps these actions can generate
+
+## Guiding Principles
+
+- Composite Actions should function as a single step or action, no matter how many steps it is composed of or how many levels of recursion it has
+- A workflow author should not need to understand the inner workings of a composite action in order to use it
+- Composite actions should leverage inputs to get values they need, they will not have full access to the `context` objects. The secrets context will **not** be available to composite actions, users will need to pass these values in as an input.
+- Other Actions should **just work** inside a composite action, without any code changes
+
+## Decisions
+
+### Composite Recursion Limit
+
+We will start with supporting a recursion limit of `3`
```

Comment:
Consider starting with a much higher number like 10 (or higher; something we feel is unreasonably high).

If i'm authoring a composite action, i shouldn't have to worry about how downstream actions are implemented. Maybe it's nodejs today, but then switches to composite. Then my workflow suddenly breaks.